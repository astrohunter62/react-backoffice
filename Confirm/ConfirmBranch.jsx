import React from 'react'
import PropTypes from 'prop-types'
import Button from 'material-ui/Button'
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog'

const ConfirmBranch = ({
  title,
  description,
  open,
  disagreeText,
  agreeText,
  onClose,
  onConfirm,
}) => (
  <Dialog
    open={open}
    onClose={onClose}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
  >
    {title ? (
      <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
    ) : null}

    <DialogContent>
      <DialogContentText id="alert-dialog-description">
        {description}
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onClose} color="primary">
        {disagreeText}
      </Button>
      <Button onClick={onConfirm} color="primary" autoFocus>
        {agreeText}
      </Button>
    </DialogActions>
  </Dialog>
)

ConfirmBranch.propTypes = {
  open: PropTypes.bool.isRequired,
  title: PropTypes.string,
  description: PropTypes.string.isRequired,
  agreeText: PropTypes.string.isRequired,
  disagreeText: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
}

export default ConfirmBranch
