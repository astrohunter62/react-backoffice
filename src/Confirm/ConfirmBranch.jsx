import React from 'react'
import PropTypes from 'prop-types'
import Button from 'material-ui/Button'
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog'
import withStyles from 'material-ui/styles/withStyles'

const styles = theme => ({
  primaryButton: {
    color: theme.palette.primary.dark,
  },
})

const ConfirmBranch = ({
  title,
  description,
  isOpen,
  disagreeText,
  agreeText,
  onClose,
  onConfirm,
  classes,
}) => (
  <Dialog
    open={isOpen}
    onClose={onClose}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
  >
    {title ? (
      <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
    ) : null}

    <DialogContent>
      <DialogContentText id="alert-dialog-description">
        {description}
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onClose} color="primary">
        {disagreeText}
      </Button>
      <Button
        onClick={onConfirm}
        color="primary"
        autoFocus
        className={classes.primaryButton}
      >
        {agreeText}
      </Button>
    </DialogActions>
  </Dialog>
)

ConfirmBranch.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  title: PropTypes.string,
  description: PropTypes.string.isRequired,
  agreeText: PropTypes.string.isRequired,
  disagreeText: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

ConfirmBranch.defaultProps = {
  title: null,
}

export default withStyles(styles)(ConfirmBranch)
