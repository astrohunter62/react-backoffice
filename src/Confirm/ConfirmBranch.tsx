import React from "react";
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from "@material-ui/core";

type ConfirmBranchProps = {
  isOpen: boolean;
  title?: string;
  description: string;
  agreeText: string;
  disagreeText: string;
  onClose: (...args: any[]) => any;
  onConfirm: (...args: any[]) => any;
  hasCloseButton: boolean;
};

const ConfirmBranch: React.SFC<ConfirmBranchProps> = ({
  title,
  description,
  isOpen,
  disagreeText,
  agreeText,
  hasCloseButton,
  onClose,
  onConfirm,
}) => (
  <Dialog
    open={isOpen}
    onClose={onClose}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
  >
    {title ? <DialogTitle id="alert-dialog-title">{title}</DialogTitle> : null}

    <DialogContent>
      <DialogContentText id="alert-dialog-description">
        {description}
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      {hasCloseButton ? (
        <Button onClick={onClose} color="primary">
          {disagreeText}
        </Button>
      ) : null}

      <Button onClick={onConfirm} color="primary" variant="contained" autoFocus>
        {agreeText}
      </Button>
    </DialogActions>
  </Dialog>
);

export default ConfirmBranch;
