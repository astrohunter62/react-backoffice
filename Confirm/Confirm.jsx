import React from 'react'
import PropTypes from 'prop-types'

import ConfirmBranch from './ConfirmBranch'

const withConfirm = Component => class Confirm extends React.Component {
  static propTypes = {
    open: PropTypes.bool,
    title: PropTypes.string,
    description: PropTypes.string.isRequired,
    agreeText: PropTypes.string,
    disagreeText: PropTypes.string,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func,
  }

  static defaultProps = {
    title: null,
    open: false,
    agreeText: 'Agree',
    disagreeText: 'Disagree',
    onClose: () => {},
  }

  constructor() {
    super()

    this.state = {
      open: false,
    }

    this.handleClose = this.handleClose.bind(this)
    this.handleConfirm = this.handleConfirm.bind(this)
  }

  componentWillReceiveProps(nextProps) {
    if (this.state.open !== nextProps.open) {
      this.setState({
        open: nextProps.open,
      })
    }
  }

  handleClose() {
    this.setState({
      open: false,
    })

    this.props.onClose()
  }

  handleConfirm() {
    this.setState({
      open: false,
    })

    this.props.onConfirm()
  }

  render() {
    return (
      <Component
        {...this.props}
        {...this.state}
        onClose={this.handleClose}
        onConfirm={this.handleConfirm}
      />
    )
  }
}

export default withConfirm(ConfirmBranch)
