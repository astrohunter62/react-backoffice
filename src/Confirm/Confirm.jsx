import React from 'react'
import PropTypes from 'prop-types'

import ConfirmBranch from './ConfirmBranch'

const withConfirm = Component => class extends React.Component {
  static propTypes = {
    open: PropTypes.bool,
    title: PropTypes.string,
    description: PropTypes.string.isRequired,
    agreeText: PropTypes.string,
    disagreeText: PropTypes.string,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func,
  }

  static defaultProps = {
    title: null,
    open: false,
    agreeText: 'Agree',
    disagreeText: 'Disagree',
    onClose: () => {},
  }

  constructor(props) {
    super(props)

    this.state = {
      open: false,
    }

    this.handleClose = this.handleClose.bind(this)
    this.handleConfirm = this.handleConfirm.bind(this)
  }

  componentWillMount() {
    this.setState({
      open: this.props.open,
    })
  }

  componentWillReceiveProps(nextProps) {
    if (this.state.open !== nextProps.open) {
      this.setState({
        open: nextProps.open,
      })
    }
  }

  handleClose() {
    this.setState({
      open: false,
    })

    this.props.onClose()
  }

  handleConfirm() {
    this.setState({
      open: false,
    })

    this.props.onConfirm()
  }

  render() {
    return (
      <Component
        {...this.props}
        {...this.state}
        onClose={this.handleClose}
        onConfirm={this.handleConfirm}
      />
    )
  }
}

export default withConfirm(ConfirmBranch)
