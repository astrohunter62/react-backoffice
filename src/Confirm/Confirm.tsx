import React from "react";
import ConfirmBranch from "./ConfirmBranch";
type WithConfirmProps = {
  isOpen?: boolean;
  title?: string;
  description: string;
  agreeText?: string;
  disagreeText?: string;
  onConfirm: (...args: any[]) => any;
  onClose?: (...args: any[]) => any;
  hasCloseButton?: boolean;
};

type WithConfirmState = {
  isOpen: boolean | undefined;
};

const withConfirm = (Component: any) =>
  class WithConfirm extends React.Component<
    WithConfirmProps,
    WithConfirmState
  > {
    constructor(props: WithConfirmProps) {
      super(props);
      this.state = {
        isOpen: false,
      };
      this.handleClose = this.handleClose.bind(this);
      this.handleConfirm = this.handleConfirm.bind(this);
    }
    componentDidMount() {
      this.setState({
        isOpen: this.props.isOpen,
      });
    }
    UNSAFE_componentWillReceiveProps(nextProps: WithConfirmProps) {
      if (this.state.isOpen !== nextProps.isOpen) {
        this.setState({
          isOpen: nextProps.isOpen,
        });
      }
    }
    handleClose() {
      this.setState({
        isOpen: false,
      });
      this.props.onClose && this.props.onClose();
    }
    handleConfirm() {
      this.setState({
        isOpen: false,
      });
      this.props.onConfirm();
    }
    render() {
      return (
        <Component
          {...this.props}
          {...this.state}
          onClose={this.handleClose}
          onConfirm={this.handleConfirm}
        />
      );
    }
  };

export default withConfirm(ConfirmBranch);
