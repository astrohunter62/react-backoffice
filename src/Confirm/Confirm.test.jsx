import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Confirm from './'

Enzyme.configure({ adapter: new Adapter() })

describe('Confirm', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Confirm
        title="Title"
        description="Desc"
        onConfirm={() => { }}
      />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('confirm is handled', () => {
    const onConfirm = jest.fn()
    const confirm = shallow((
      <Confirm
        isOpen
        title="Title"
        description="Desc"
        onConfirm={onConfirm}
      />
    ))

    expect(confirm.state().isOpen).toBe(true)
    confirm.instance().handleConfirm()

    expect(onConfirm).toHaveBeenCalled()
  })

  it('close is handled', () => {
    const onClose = jest.fn()
    const confirm = shallow((
      <Confirm
        isOpen
        title="Title"
        description="Desc"
        onConfirm={() => {}}
        onClose={onClose}
      />
    ))

    expect(confirm.state().isOpen).toBe(true)
    confirm.instance().handleClose()

    expect(onClose).toHaveBeenCalled()
  })

  it('changes open', () => {
    const confirm = shallow((
      <Confirm
        isOpen
        description="Desc"
        onConfirm={() => { }}
      />
    ))

    confirm.setProps({
      isOpen: false,
    })
  })
})
