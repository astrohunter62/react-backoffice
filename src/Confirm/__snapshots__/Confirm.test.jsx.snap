// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm renders correctly 1`] = `""`;
